packages/amplify-cli/src/commands/diagnose.ts

```diff
@@ -240,6 +240,6 @@ const hashedProjectIdentifiers = (): { projectIdentifier: string; projectEnvIden
 };
 
 const getAppId = (): string => {
-  const meta = stateManager.getMeta();
-  return _.get(meta, ['providers', 'awscloudformation', 'AmplifyAppId']);
+  const meta = stateManager.getMeta(undefined, { throwIfNotExist: false });
+  return meta?.providers?.awscloudformation?.AmplifyAppId ?? 'noAppIdFound';
```

Comment:
Are we checking this some where 'noAppIdFound' ?

Reply:
It's used to create the projectIdentifier: https://github.com/aws-amplify/amplify-cli/pull/11513/files#diff-71cabcaa3379af3b9458f6802b25c65f6c00768606d5ef7f8c010094d8720aa1R238-R239

But there's no `amplify-meta.json` file when no backend is pulled, so the existing code always fails on ln 243.